File: accounts/src/main/java/com/cinelist/ms/accounts/database/repositories/UserRepository.java
package com.cinelist.ms.accounts.database.repositories;

import com.cinelist.ms.accounts.database.models.User;
import org.springframework.data.jpa.repository.JpaRepository;

import java.util.Optional;
import java.util.UUID;

public interface UserRepository extends JpaRepository<User, UUID> {
    Optional<User> findByIdentifier(UUID identifier);
    Optional<User> findByUsername(String username);
    Optional<User> findByEmail(String email);
}

File: accounts/src/main/kotlin/com/cinelits/ms/accounts/database/repositories/UserRepository.kt
package com.cinelits.ms.accounts.database.repositories

import com.cinelits.ms.accounts.database.models.User
import org.springframework.data.jpa.repository.JpaRepository
import java.util.Optional
import java.util.UUID

interface UserRepository : JpaRepository<User, UUID>{
    fun findByIdentifier(identifier: UUID): Optional<User>
    fun findByUsername(username: String?): Optional<User>
    fun findByEmail(email: String?): Optional<User>
}